### PR TITLE
Lots of updates and cleanup

### PR DESCRIPTION
**Pull request by TacoTheDank**
Welcome to oh-my-god-this-is-a-massive-pull-request, here are the changes:

### 1st commit
- Deleted generated and obsolete files

### 2nd commit
- Updated the Gradle wrapper 5.4.1 -> 5.6.2

### 3rd commit
- Cleaned up the build.gradle files a bit
- Added changelogs to dependencies
- Updated Guava 24.1-jre -> 24.1.1-jre

### 4th commit
- Added an .editorconfig for codestyle consistency

### 5th commit
- Automatically adjusted the codestyle to the .editorconfig using Ctrl+Alt+L

### 6th commit
- Did an automatic code cleanup using lint

### 7th commit (here's the meaty bit)
- Removed some redundant code
- Take advantage of Java 1.8 lambdas and other features
- Optimized some code
- Narrowed down some access modifiers
- Converted a bunch of variables to local variables
- Turn some unused methods into voids

### 8th commit
- Basically commit 7, but for the XML files
- Fix some strings, convert ellipses into the ellipses symbol

### 9th commit
- Updated libraries:
  - Glide 4.9.0 -> 4.10.0
  - Picasso 2.4.0 -> 2.5.2
  - com.github.rey5137:material 1.2.1 -> 1.3.0
  - Guava 24.1.1-jre -> 28.1-jre
  - Gson 2.8.2 -> 2.8.6
  - Butterknife 10.0.0 -> 10.2.0
  - Material-Dialogs 0.8.5.8 -> 0.8.6.0
- Converted TransitionsEverywhere into AndroidX Transition (reason is TE 2.x now uses AndroidX's transition library instead of having every AndroidX transition class in TE's library, and there's only one usage, so we can delete this library)
- Deleted unused libraries (commons-codec, rippleeffect, transitionseverywhere (now unused because of above))

### 10th commit
- Fixed two things that were causing build errors

### CODE REVIEW

**Comment by Perflyst** [comment]
Since this is mostly automatic cleanup and refactoring in Android Studio and I did not found "new" bugs while testing the app with these changes I am so free and do not through all the changed files.
